protocol/communities/manager.go

```diff
@@ -83,7 +90,7 @@ func (t *HistoryArchiveDownloadTask) Cancel() {
 	close(t.CancelChan)
 }
 
-func NewManager(identity *ecdsa.PrivateKey, db *sql.DB, encryptor *encryption.Protocol, logger *zap.Logger, verifier *ens.Verifier, accountsManager *account.GethManager, transport *transport.Transport, torrentConfig *params.TorrentConfig) (*Manager, error) {
+func NewManager(identity *ecdsa.PrivateKey, db *sql.DB, encryptor *encryption.Protocol, logger *zap.Logger, verifier *ens.Verifier, accountsManager *account.GethManager, tokenManager *token.Manager, transport *transport.Transport, torrentConfig *params.TorrentConfig) (*Manager, error) {
```

Comment:
I notice that this function has a very long signature. I wonder if this is a sign that the function is growing too complex or the method of interfacing it needs to be simplified.

See Go style guide on this subject https://google.github.io/styleguide/go/best-practices.html#function-argument-lists . One potential solution is to use an Option Structure see https://google.github.io/styleguide/go/best-practices.html#option-structure.

In fact `NewMessenger` already uses a variadic parameter for Option structs.